pkg/agent/utils.go

```diff
@@ -362,6 +362,12 @@ func GetOrderedKubeletConfigFlagString(config *datamodel.NodeBootstrappingConfig
 	sort.Strings(keys)
 	var buf bytes.Buffer
 	for _, key := range keys {
+		/* Override --cluster-dns if LocalDNS is enabled for the agentpool.
+		If localdns is enabled for the agentpool,
+		then Kubelet's --cluster-dns flag should be pointed to localdns clusterlistenerIP address. */
+		if key == "--cluster-dns" && profile.ShouldEnableLocalDNS() {
```

Comment:
I am a little bit confused about this, who is passing "--cluster-dns" config in? is it from RP vmss.go

Reply:
Set this value in Default(), and validate the value in Validate()